store/src/java/com/zimbra/cs/mailbox/Message.java

```diff
@@ -1290,6 +1301,47 @@ private void getInfoForAssociatedCalendarItem(Account acct, Invite cur, String m
         }
     }
 
+    public boolean hasDataSourceId() {
+        return !Strings.isNullOrEmpty(dsId);
+    }
+
+    public String getDataSourceId() {
+        return dsId;
+    }
+
+    public boolean isSentByMe() {
+        return sentByMe;
+    }
+
+    public EventFlag getEventFlag() {
+        return eventFlag;
+    }
+
+    void setEventFlag(EventFlag eventFlag) {
+        this.eventFlag = eventFlag;
+        try {
+            saveMetadata();
+        } catch (ServiceException e) {
+            ZimbraLog.event.warn("unable to save metadata; duplicate events may be generated");
```

Comment:
Worth including the exception?  Could make it dependant on say event.isDebugEnabled() if want to keep noise level down.